### PR TITLE
fix `For element's text properties we should convert non-string values to string` (close #1091)

### LINKED ISSUE #1091: For element's text properties we should convert non-string values to string
E. g. 
```js
var value = { a: 1 };
document.body.innerText = value;
```
We should do: `String({a:1})`  before processing and assigning.

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @helen-dikareva @miherlosev

### CODE REVIEW

File: src/client/sandbox/code-instrumentation/properties/index.js [hunk 1]
```diff
@@ -274,16 +274,18 @@ export default class PropertyAccessorsInstrumentation extends SandboxBase {
                     var isStyleEl  = domUtils.isStyleElement(el);
                     var isScriptEl = domUtils.isScriptElement(el);
 
-                    if (value) {
+                    var processedValue = String(value);
```
Comment (miherlosev) [inline src/client/sandbox/code-instrumentation/properties/index.js]: Don't need the new variable.
Just `value=String(value)`.

File: test/client/fixtures/sandbox/code-instrumentation/setters-test.js [hunk 2]
```diff
@@ -246,28 +246,41 @@ test('innerHTML, innerText, text, textContent', function () {
     var testProperties      = ['innerHTML', 'innerText', 'text', 'textContent'];
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/code-instrumentation/setters-test.js]: Add `outerHtml` checking to the `testProperties`

File: src/client/sandbox/code-instrumentation/properties/index.js (hunk 1 above)
Comment (LavrovArtem) [inline src/client/sandbox/code-instrumentation/properties/index.js]: We need to return origin value.
\> script.innerHTML = {a: 1}; // return object {a: 1}
\> script.innerHTML === '[object Object]' // true
[see test](https://github.com/DevExpress/testcafe-hammerhead/pull/1092/files/a80cab1cf06b72e4f67393f84a2eccf0ccf985e3#diff-bbfd287e468ceb547f28a177a62cb879R280)

Comment (miherlosev) [inline src/client/sandbox/code-instrumentation/properties/index.js]: Ok

File: test/client/fixtures/sandbox/code-instrumentation/setters-test.js (hunk 2 above)
Comment (LavrovArtem) [inline test/client/fixtures/sandbox/code-instrumentation/setters-test.js]: [separate test](https://github.com/DevExpress/testcafe-hammerhead/pull/1092/files#diff-bbfd287e468ceb547f28a177a62cb879R355)

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
FPR

File: src/client/sandbox/code-instrumentation/properties/index.js [hunk 3]
```diff
@@ -274,16 +274,18 @@ export default class PropertyAccessorsInstrumentation extends SandboxBase {
                     var isStyleEl  = domUtils.isStyleElement(el);
                     var isScriptEl = domUtils.isScriptElement(el);
 
-                    if (value) {
+                    var processedValue = value !== null && value !== void 0 ? String(value) : value;
```
Comment (helen-dikareva) [inline src/client/sandbox/code-instrumentation/properties/index.js]: I think it's not quite right check. When I try to set `null` to `innerHTML`:
```text
document.body.innerHTML = null
null
document.body.innerHTML
""
```
But if I try to set `undefined`:
```text
document.body.innerHTML = void 0
undefined
document.body.innerHTML
"undefined"
```
We should take this into account.

File: src/client/sandbox/code-instrumentation/properties/index.js [hunk 4]
```diff
@@ -274,16 +274,18 @@ export default class PropertyAccessorsInstrumentation extends SandboxBase {
...
+                    if (processedValue) {
                         if (isStyleEl)
-                            value = styleProcessor.process('' + value, urlUtils.getProxyUrl, true);
+                            processedValue = styleProcessor.process(processedValue, urlUtils.getProxyUrl, true);
                         else if (isScriptEl)
-                            value = processScript('' + value, true);
-                        else if (value !== null)
-                            value = processHtml('' + value, el.tagName);
+                            processedValue = processScript(processedValue, true);
+                        else if (processedValue !== null)
```
Comment (helen-dikareva) [inline src/client/sandbox/code-instrumentation/properties/index.js]: Now this check is redundant

File: src/client/sandbox/code-instrumentation/properties/index.js [hunk 5]
```diff
@@ -601,16 +607,18 @@ export default class PropertyAccessorsInstrumentation extends SandboxBase {
                 get: el => PropertyAccessorsInstrumentation.removeProcessingInstructions(el.textContent),
 
                 set: (el, text) => {
-                    if (text) {
+                    var processedText = text !== null && text !== void 0 ? String(text) : text;
```
Comment (helen-dikareva) [inline src/client/sandbox/code-instrumentation/properties/index.js]: May be exclude this part of code  into method
```js
var processedText = text !== null && text !== void 0 ? String(text) : text;

if (processedText) {
    if (domUtils.isScriptElement(el))
        el.textContent = processScript(processedText, true);
    else if (domUtils.isStyleElement(el))
         el.textContent = styleProcessor.process(processedText, urlUtils.getProxyUrl, true);
    else
         el.textContent = processedText;
}
else
    el.textContent = processedText;
```

Comment (LavrovArtem) [inline src/client/sandbox/code-instrumentation/properties/index.js]: agree

**Pull request merged**